### PR TITLE
Issue #544: add test ci in github workflow

### LINKED ISSUE #544: Move CI to GitHub actions
## Subject of the issue

- https://blog.travis-ci.com/2020-11-02-travis-ci-new-billing
- https://ropensci.org/technotes/2020/11/19/moving-away-travis/

As explained in the links above, Travis CI is changing their pricing plan, we need to check in more details
- whether it will impact us considering our current CI activities
- does GitHub action can do the same job and is it easy to move? If yes, maybe better to move earlier than later

### PR DESCRIPTION
**Pull request by mathpluscode**
# Description

Use github workflow for CI

Fixes #544

## Type of change

What types of changes does your code introduce to DeepReg?

_Please check the boxes that apply after submitting the pull request._

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation Update (fix or improvement on the documentation)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Other (if none of the other choices apply)

## Checklist

_Please check the boxes that apply after submitting the pull request._

_If you're unsure about any of them, don't hesitate to ask. We're here to help! This is
simply a reminder of what we are going to look for before merging your code._

- [x] I have
      [installed pre-commit](https://deepreg.readthedocs.io/en/latest/contributing/setup.html)
      using `pre-commit install` and formatted all changed files. If you are not
      certain, run `pre-commit run --all-files`.
- [x] My commits' message styles matches
      [our requested structure](https://deepreg.readthedocs.io/en/latest/contributing/commit.html),
      e.g. `Issue #<issue number>: detailed message`.
- [x] I have updated the
      [change log file](https://github.com/DeepRegNet/DeepReg/blob/main/CHANGELOG.md)
      regarding my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by mathpluscode** [comment]
:) migration is done, the project code cov is ok, @YipengHu @NMontanaBrown could you have a look? This is blocking all other works

**Review by YipengHu — APPROVED** [review]

**Comment by mathpluscode** [comment]
@acasamitjana if you have time to check this :) as without this PR, no one else can be done

File: README.md [hunk 1]
```diff
@@ -39,9 +39,6 @@
       <a href="https://github.com/psf/black">
       <img src="https://img.shields.io/badge/code%20style-black-000000.svg" alt="Cod Style">
       </a>
-      <a href="https://travis-ci.com/DeepRegNet/DeepReg">
```
Comment (zacbaum) [inline README.md:42]: Is there any other way we can display build status now? I think this is important to have on the README.

File: docs/source/contributing/test.md [hunk 2]
```diff
@@ -4,8 +4,7 @@ In DeepReg, we use [pytest](https://docs.pytest.org/en/stable/) (not
 [unittest](https://docs.python.org/3/library/unittest.html)) for unit tests to ensure a
 certain code quality and to facilitate the code maintenance.
 
-The testing is checked via [Travis-CI](https://travis-ci.com/github/DeepRegNet/DeepReg)
```
Comment (zacbaum) [inline docs/source/contributing/test.md:7]: We should still indicate *where* testing is done; would be best to indicate that a GitHub action/workflow is used for this (and perhaps a link to relevant information).

**Review by zacbaum — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (NMontanaBrown) [reply README.md:42]: Agree

Comment (mathpluscode) [reply README.md:42]: Maybe but I will do that later, as I didn't find out a good way to do so for now.

File: docs/source/contributing/test.md (hunk 2 above)
Comment (mathpluscode) [reply docs/source/contributing/test.md:7]: ok

File: README.md (hunk 1 above)
Comment (zacbaum) [reply README.md:42]: @mathpluscode see:

https://andresand.medium.com/adding-github-action-workflow-status-badge-to-your-repository-22ccea025af6

or

https://docs.github.com/en/free-pro-team@latest/actions/managing-workflow-runs/adding-a-workflow-status-badge

**Comment by mathpluscode** [comment]
@zacbaum thx to your example! I've added the badges!

<a href="https://github.com/DeepRegNet/DeepReg/actions?query=workflow%3A%20Unit%20Test">
Unit Test
</a>
<a href="https://github.com/DeepRegNet/DeepReg/actions?query=workflow%3A%20Demo%20Test">
Demo Test
</a>

**Review by zacbaum — APPROVED** [review]
Thanks @mathpluscode! This looks great 😄

**Pull request merged**